### PR TITLE
refactor lib/private/OCS

### PR DESCRIPTION
**Pull request by shdehnavi**
## Summary
The required adjustments have been made to the classes in `/lib/private/OCS ` namespace.

The improvements:

- Using PHP8's constructor property promotion
- Adding return types
- Adding types to properties
- Replacing qualifiers with imports

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Nice

File: lib/private/OCS/Result.php [hunk 1]
```diff
@@ -32,31 +32,32 @@
 
 class Result {
 	/** @var array  */
```
Comment (fsamapoor) [inline lib/private/OCS/Result.php:34]:
```suggestion
```

Comment (fsamapoor) [reply lib/private/OCS/Result.php:34]: Now that the properties are typed, I think there is no value in keeping the doc blocks and you can remove them as well.

**Review by ChristophWurst — APPROVED** [review]
Please squash into one commit

**Comment by shdehnavi** [comment]
@ChristophWurst All commits squashed into one commit and all conflicts have been resolved.

**Comment by ChristophWurst** [comment]
> Error: lib/private/OCS/Provider.php:42:3: UndefinedThisPropertyAssignment: Instance property OC\OCS\Provider::$appManager is not defined (see https://psalm.dev/040)
> Error: lib/private/OCS/Provider.php:60:7: UndefinedThisPropertyFetch: Instance property OC\OCS\Provider::$appManager is not defined (see https://psalm.dev/041)

**Comment by shdehnavi** [comment]
@ChristophWurst Psalm errors have been addressed in the last commit.
Also, All commits are squashed into one commit.

**Review by ChristophWurst — APPROVED** [review]

**Review by fsamapoor — APPROVED** [review]

**Pull request merged**